test/e2e/test_tools.py

```diff
@@ -24,16 +24,17 @@ class TestTools(unittest.TestCase):
     test_project_minimal_file = 'testdata/test-project-minimal.json'
     test_data_file = 'testdata/test-data-systematic.xml'
 
-    def setUp(self) -> None:
```

Comment:
This has nothing to do with the core of this PR, but I came across it. It's not necessary to do the setup and teardown before and after each method call.

Reply:
I remember that I've had problems without emptying the lists before and after the tests but if it works now without, then the better.